src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -1099,6 +1099,50 @@ void WorldSession::HandlePlayerLoginFromDB(LoginQueryHolder* holder)
         pCurrChar->CheckAllAchievementCriteria();
     }
 
+        // Reputations if "StartAllReputation" is enabled, -- TODO: Fix this in a better way
+    if (sWorld->getBoolConfig(CONFIG_START_ALL_REP))
+    {
+        ReputationMgr& repMgr = pCurrChar->GetReputationMgr();
```

Comment:
Not relevant to your change but since you did touch it, the factions should be stored in arrays and then looped through for all 3 sets of setting factions.

Reply:
the code is not from me, dunno who will rework that code, it will probably be forgotten forever :P